workspace/resource_directory.go

```diff
@@ -0,0 +1,56 @@
+package workspace
+
+import (
+	"context"
+	"errors"
+
+	"github.com/databrickslabs/terraform-provider-databricks/common"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// ResourceDirectory manages directories
+func ResourceDirectory() *schema.Resource {
+	s := map[string]*schema.Schema{
+		"path": {
+			Type:     schema.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+		"delete_recursive": {
+			Type:     schema.TypeBool,
+			Default:  false,
+			Optional: true,
+		},
+	}
+
+	return common.Resource{
+		Schema:        s,
+		SchemaVersion: 1,
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			notebooksAPI := NewNotebooksAPI(ctx, c)
+			path := d.Get("path").(string)
+			if err := notebooksAPI.Mkdirs(path); err != nil {
+				// TODO: handle RESOURCE_ALREADY_EXISTS
```

Comment:
For directory you probably have to remove this todo

Reply:
I think it still needs to be handled. If we try to create the path `/a/b/c` but `a/b` is already a notebook or other type of object it will fail.

![image](https://user-images.githubusercontent.com/14082115/122565656-1af7ab80-d01d-11eb-8972-072136fb7ecf.png)